tensorflow/java/src/gen/java/org/tensorflow/processor/operator/OperatorProcessor.java

```diff
@@ -0,0 +1,115 @@
+package org.tensorflow.processor.operator;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic.Kind;
+
+public final class OperatorProcessor extends AbstractProcessor {
+
+  @Override
+  public SourceVersion getSupportedSourceVersion() {
+    return SourceVersion.latestSupported();
+  }
+
+  @Override
+  public synchronized void init(ProcessingEnvironment processingEnv) {
+    super.init(processingEnv);
+    messager = processingEnv.getMessager();
+    filer = processingEnv.getFiler();
+  }
+
+  @Override
+  public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    // Nothing needs to be done at the end of all rounds.
+    if (roundEnv.processingOver()) {
+      return false;
+    }
+
+    // Collect all classes tagged with our annotation.
+    Set<TypeElement> opClasses = new HashSet<TypeElement>();
+    for (TypeElement annotation : annotations) {
```

Comment:
Shouldn't there be precisely one `annotations`? If so, perhaps we can assert that `annotations.size()` is 1 and then call `collectOpClasses` once?

Reply:
It can be zero or 1 depending on which phase this runs in. I'll combine this with the rest of the constraint reformulation you suggested. Ptal, and if overall it reads better to you, sgtm.